### PR TITLE
Decoupled Multi-currency module

### PR DESCRIPTION
**Pull request by ricardo**
Context: pcrvl0-1iw-p2

#### Changes proposed in this Pull Request

The purpose of this PR is to merge the feature branch `multi-currency-v2` into `develop`. All of the PRs that were previously merged into multi-currency-v2 were tested (pcrvl0-1oD-p2) and code reviewed (36-gh-Automattic/woopayments).

#### Testing instructions

- Switch to this branch.
- Run `composer dump-autoload`.
- Run `npm run build:client`.
- Smoke test multi-currency and single currency settings and a multi-currency transaction.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9617` or branch name `multi-currency-v2` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=multi-currency-v2) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 786277c07406b876fa3c05526ab78d419000c843
- Build time: 2024-10-25 03:46:55 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by reykjalin — APPROVED** [review]
Looks good and tests well! Great work 👏

File: changelog/fix-backwards-compat-mccy-instance [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: fix
+
+Provide backwards-compatible method for retrieving the multi-currency instance.
```
Comment (cesarcosta99) [inline changelog/fix-backwards-compat-mccy-instance:4]: I wanted to highlight we have two changelog entries in this PR. If that's intentional, no worries, but I thought it'd be good to bring it up.

**Review by cesarcosta99 — APPROVED** [review]
Code looks good and tests well! 🚀 See my comment about the changelog before merging, but it's not something concerning.

File: changelog/fix-backwards-compat-mccy-instance (hunk 1 above)
Comment (ricardo) [reply changelog/fix-backwards-compat-mccy-instance:4]: It's intentional. Thanks for flagging. 👍

**Comment by ricardo** [comment]
@cesarcosta99 I synced [6792d84](https://github.com/Automattic/woocommerce-payments/pull/9617/commits/6792d84244801cd0b6537d11cd066fedc0c19de8) now in this PR. Could you please take a look at this commit before I merge?

**Review by cesarcosta99 — APPROVED** [review]
6792d84 looks good and tests well 👍 Even though it has many LOC changed the changes are straightforward.

- Smoke test multi-currency and single currency settings and a multi-currency transaction ✅ 
- I searched for old references of `gateway_context` and didn't find any ✅

**Pull request merged**